networkdb/networkdb.go

```diff
@@ -627,6 +631,12 @@ func (nDB *NetworkDB) JoinNetwork(nid string) error {
 		logrus.Errorf("Error bulk syncing while joining network %s: %v", nid, err)
 	}
 
+	// Mark the network as being synced
+	// note this is a best effort, we are not checking the result of the bulk sync
+	nDB.Lock()
+	n.inSync = true
+	nDB.Unlock()
```

Comment:
this flag is mainly to reduce the number of messages sent around during the join network. The current logic tries to improve convergence readvertising also messages that are coming from tcp sync. Before this change the behavior was such that if the table had 10K entries the node was starting with 10K messages in the queue because was trying to propagate all that entries that came to know during the join network tcp sync. This change instead make sure that during the first tcp sync no entries are being resent around. This way a node will join and will have close to 0 entries in the queue and full state synced.

Now to answer your question if the bulkSync succeed with at least 1 node, that will be enough to have the internal state "enough in sync" in term of entries, because in theory all the nodes on the same network holds the same content. The difference will only be on the entries still "in flight", but the number of them should be negligible compared with the number of entries coming from the bulk sync.
This flag does not really affect convergence logic because anyway this node when it joins start with an empty table and empty queue, so was already not helping in propagating the messages around.

Reply:
let me know if I addressed the question properly

Reply:
yes, that explains it!